src/http/client.rs

```diff
@@ -37,11 +37,11 @@ impl AybClient {
     ) -> Result<T, AybError> {
         let status = response.status();
         match status {
-            status if status == expected_status => response.json::<T>().await.or_else(|err| {
-                Err(AybError {
+            status if status == expected_status => {
```

Comment:
Sanity-checking that this line is equivalent to the original. We're stripping the `Err` --- is that OK?

Reply:
Yes because it is being replaced with a [`map_err`](https://doc.rust-lang.org/std/result/enum.Result.html#method.map_err).
That matching is kinda weird anyways, I'll take a look to it...

Reply:
I've updated the implementation to a more "conventional" one using map and map_err, unless I'm mistaken they are equivalent